### PR TITLE
Fix branch automatically not linking on quick pull request

### LINKED ISSUE #691: Automatically link branch in "Open a pull request"
Open a pull request

The `travis-publish` branch isn't clickable.

Example URL:

```
https://github.com/tanmayrajani/notifications-preview-github/compare/master...compact?quick_pull=1
```

### PR DESCRIPTION
**Pull request by tanmayrajani**
Fixes #691

### CODE REVIEW

**Comment by bfred-it** [comment]
I'm not entirely sure but I think this should use the existing linkifyBranch function

**Comment by tanmayrajani** [comment]
Yes, `linkifyBranchRefs` function indeed does similar things. But the problems with re-using that are:
- the selector for the branch name on quick PR page (`.branch-name`) is different than the ones used there
- that function makes use of `title` property of the branch-name element which is absent in this case
- the things that function does/checks for (i.e. deleted branch/unknown repo) are not applicable in this use case as per my understanding.. but I could be wrong.

Let me know if I misunderstood anything here!

**Comment by bfred-it** [comment]
You're right! The DOM is completely different.

File: src/libs/page-detect.js [hunk 1]
```diff
@@ -46,6 +46,8 @@ export const isCommit = () => isSingleCommit() || isPRCommit() || (isPRFiles() &
 
 export const isCompare = () => isRepo() && /^\/compare/.test(getRepoPath());
 
+export const isQuickPR = () => isCompare() && /\?quick_pull=1$/.test(location.href);
```
Comment (bfred-it) [inline src/libs/page-detect.js]: Perhaps the second part should be:

```js
/[?&]quick_pull=1(&|$)/.test(location.search)
```

Comment (tanmayrajani) [reply src/libs/page-detect.js]: Right! Did so in additional commit, thanks!

File: src/libs/page-detect.js [hunk 2]
```diff
@@ -46,6 +46,8 @@ export const isCommit = () => isSingleCommit() || isPRCommit() || (isPRFiles() &
 
 export const isCompare = () => isRepo() && /^\/compare/.test(getRepoPath());
 
+export const isQuickPR = () => isCompare() && /[?&]quick_pull=1(&|$)/.test(location.search);
```
Comment (hkdobrev) [inline src/libs/page-detect.js]: It's funny that `?expand=1` has a different copy and no such branch link in the DOM.

File: test/page-detect.js [hunk 3]
```diff
@@ -192,6 +192,15 @@ test('isCompare', urlMatcherMacro, pageDetect.isCompare, [
 	'https://github.com/sindresorhus/refined-github/graphs'
 ]);
 
+test('isQuickPR', urlMatcherMacro, pageDetect.isQuickPR, [
+	'https://github.com/sindresorhus/refined-github/compare/master...branch-name?quick_pull=1',
+	'https://github.com/sindresorhus/refined-github/compare/branch-1...branch-2?quick_pull=1'
```
Comment (hkdobrev) [inline test/page-detect.js]: Could you please add positive tests with a single branch (where the default base branch is implied)?

File: test/page-detect.js [hunk 4]
```diff
@@ -192,6 +192,15 @@ test('isCompare', urlMatcherMacro, pageDetect.isCompare, [
...
+], [
+	'https://github.com/sindresorhus/refined-github',
+	'https://github.com/sindresorhus/refined-github/compare',
+	'https://github.com/sindresorhus/refined-github/compare/'
```
Comment (hkdobrev) [inline test/page-detect.js]: Could you please add negative tests with compare URLs with one branch, two branches and also `?expand=1` which also opens a new PR from a compare URL, but is not a quick PR URL?

**Pull request merged**